### PR TITLE
Please check homework4

### PR DESCRIPTION
**Pull request by DariaLoza**

### CODE REVIEW

File: tests/tests_homework4/test_task1.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def test_read_magic_number_true():
+    """_Testing True result_"""
+    test_file_1 = os.open("file_1.txt", os.O_CREAT | os.O_WRONLY)
+    number = "2"
+    encoded_number = str.encode(number)
+    os.write(test_file_1, encoded_number)
+    path = os.path.abspath("file_1.txt")
+    if os.path.isfile(path):
```
Comment (icanhardcode) [inline tests/tests_homework4/test_task1.py]: you should write this check inside your function ```read_magic_number```

File: homeworks/homework4/task2.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+https://docs.python.org/3/library/urllib.request.html#urllib.request.urlopen
+"""
+import urllib.request as request
+
+
+def count_dots_on_i(url):
+    count_i = 0
+    with request.urlopen(url) as response:
+        for element in response.read().decode("utf-8"):
+            if element == "i":
```
Comment (icanhardcode) [inline homeworks/homework4/task2.py]: look at built-in methods for ```str``` in python

File: homeworks/homework4/task4.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ - how to write doctests
+ - how to run doctests
+assert fizzbuzz(5) == ["1", "2", "fizz", "4", "buzz"]
+https://en.wikipedia.org/wiki/Fizz_buzz
+** Энциклопедия профессора Фортрана page 14, 15, "Робот Фортран, чисть картошку!"
+"""
+from typing import List
+
+
+def fizzbuzz(n: int) -> List[str]:
```
Comment (icanhardcode) [inline homeworks/homework4/task4.py:20]: works incorrect

File: tests/tests_homework4/test_task4.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+from homeworks.homework4.task4 import fizzbuzz
+
+
+def test_fizzbuzz_function_with_5():
```
Comment (icanhardcode) [inline tests/tests_homework4/test_task4.py:4]: add tests for bigger numbers

File: homeworks/homework4/task3.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+You will learn:
+ - how to write to stderr
+ - how to test output to the stderr and stdout
+
+"""
+
+import sys
+
+
+def my_precious_logger(text: str):
```
Comment (nikitosboroda) [inline homeworks/homework4/task3.py:23]: works incorrect

**Review by icanhardcode — CHANGES_REQUESTED** [review]

File: homeworks/homework4/task1.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def read_magic_number(file_path) -> bool:
+    with open(file_path, "r") as file:
+        path = os.path.abspath(file_path)
+        if os.path.isfile(path):
+            first_line = file.readline()
+            if not first_line.isdigit():
+                raise ValueError
+            else:
```
Comment (icanhardcode) [inline homeworks/homework4/task1.py:36]: can be rewritten

**Review by icanhardcode — CHANGES_REQUESTED** [review]

**Review by icanhardcode — APPROVED** [review]

**Pull request merged**